### PR TITLE
Support any CSS length in `mobile-breakpoint`

### PR DESCRIPTION
**Pull request by LeaVerou**

### CODE REVIEW

File: src/components/page/page.ts [hunk 1]
```diff
@@ -138,17 +140,21 @@ export default class WaPage extends WebAwesomeElement {
   @property({ attribute: 'nav-open', reflect: true, type: Boolean }) navOpen = false;
 
   /**
-   * At what "px" to hide the "menu" slot and collapse into a hamburger button
+   * At what page width to hide the "menu" slot and collapse into a hamburger button.
+   * Accepts both numbers (interpreted as px) and CSS lengths (e.g. `50em`), which are resolved based on the root element.
```
Comment (KonnorRogers) [inline src/components/page/page.ts]:
```suggestion
   * At what page width to hide the "navigation" slot and collapse into a hamburger button.
   * Accepts both numbers (interpreted as px) and CSS lengths (e.g. `50em`), which are resolved based on the root element.
```

Little nitpicky, but in talking with @lindsaym-fa it is a little confusing that the menu isn't technically what collapses, but the "navigation" slot that collapses. (turns from the `<nav>` to a `<dialog>`)

File: src/internal/css-values.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  if (resolved?.endsWith('px')) {
+    return parseFloat(resolved);
+  }
+
+  return Number(resolved);
+}
+
+export function toLength(px: number | string): string {
+  return Number.isNaN(Number(px)) ? `${px}px` : (px as string);
```
Comment (KonnorRogers) [inline src/internal/css-values.ts]: is this backwards?? Or am I misunderstanding the logic?

If you pass a `number` you should get back `${number}px` if you pass anything else, we just use the string.

```suggestion
  return Number.isNaN(Number(px)) ? (px as string) : `${px}px`;
```

Am I missing something obvious?

File: src/internal/css-values.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  { on = document.documentElement, as = 'length', initialValue = initialValues[as] }: ResolveOptions = {}
+) {
+  const resolver = `--wa-${as}-resolver`;
+
+  if (!window.CSS || !CSS.registerProperty) {
+    return value;
+  }
+
+  if (!definedProperties.has(resolver)) {
+    CSS.registerProperty({
```
Comment (KonnorRogers) [inline src/internal/css-values.ts:28]: Support on this was way larger than I expected 👀 

https://caniuse.com/?search=registerProperty

**Review by KonnorRogers — APPROVED** [review]
This is insanely clever! Thanks for making this happen. A couple of questions, otherwise good to go from me!

File: src/internal/css-values.ts (hunk 2 above)
Comment (LeaVerou) [reply src/internal/css-values.ts]: Yikes, yes it is. Now I question my testing 🤣

**Pull request merged**